Services/Contact/classes/class.ilMailingLists.php

```diff
@@ -144,6 +144,25 @@ public function deleteTemporaryLists(): void
         }
     }
 
+    /**
+     * @return ilMailingList[]
+     */
+    public function getAssigned(): array
+    {
+        $res = $this->db->queryF(
+            'SELECT * FROM addressbook_mlist_ass as ass INNER JOIN addressbook_mlist as main ON main.ml_id = ass.ml_id WHERE usr_id = %s',
+            [ilDBConstants::T_INTEGER],
+            [$this->user->getId()]
+        );
+
+        $entries = [];
+        while ($row = $res->fetchRow(ilDBConstants::FETCHMODE_OBJECT)) {
+            $entries[] = new ilMailingList(new ilObjUser((int) $row->user_id), $row->ml_id);
```

Comment:
```suggestion
            $entries[] = new ilMailingList(new ilObjUser((int) $row->user_id), (int) $row->ml_id);
```

Reply:
Creating user instances here is very expensive in terms of performance. We should avoid this in scenarios where dozens of user accounts can potentially be deleted at once.

Please use simple SQL statements within `ilMailingLists`.